### ISSUE TITLE
Save / Load method in simulation class

### ISSUE THREAD

**Issue by AmosEgel** [disordered-photonics/celes#12]
Up to now, CELES simulations were saved by trivially saving the celes_simulation object as a whole. This seems broken since recent updates in the data structure (Lorenzo, is that correct?).
Another drawback of saving the whole celes_simulation object is that lookup tables and other heavy stuff might be saved leading to large file sizes.

To meet this issue, we could implement a save method in the simulation class that manually stores the relevant attributes into some data structure which is then saved, as well as a load method that reads from that structure and initializes a simulation object.

**Comment by lpattelli** [comment]
Yes, everything is now subclassed from the matlab.System superclass in order to implement a property-value input and one-time calculations inside setupImpl. The constructor calls the one-time methods, which in turn requires that all input parameters have been passed properly at construction time.

I've added two standalone functions to dump few relevant fields of a simulation object to a struct and instantiate a simulation object reading from one such struct. probably this is not the most elegant way to go, but provides a good flexibility to include only the relevant variables and don't bloat the dump file. the correct approach would be probably to specify custom saveobj/loadobj or saveObjImpl/loadObjImpl methods, but I didn't succeed to make them work.

To make these functions meaningful I guess that the next step is to manually store the content of celes_tables as well, and then change the run method to check whether these tables are already set so to use their content instead of recalculating everything from scratch.

**Comment by AmosEgel** [comment]
Why saving celes_tables? I assume they cost relatively much disc space, and it doesn't take much time to compute them once.

**Comment by AmosEgel** [comment]
Oh, right - the scattered field coefficients are in the `celes_tables`. Yes, saving between the iterations and then resuming a cancelled simulation is a good idea and should be possible.
